jupyterhub_moss/spawner.py

```diff
@@ -433,14 +418,12 @@ def __update_spawn_commands(self, cmd_path):
         self.cmd = [os.path.join(cmd_path, "jupyterhub-singleuser")]
 
     async def start(self):
-        partitions_info = await self._get_partitions_info()
-        partition_info = partitions_info[self.user_options["partition"]]
-
-        # Exceptions raised by the checks are catched by the caller, and
-        # a "500 Internal Server Error" is returned to the frontend.
-        self.__check_user_options(partition_info)
+        # Specific handling of landing URL (e.g., to start jupyterlab)
+        self.default_url = self.user_options.get("default_url", "")
+        self.log.info(f"Used default URL: {self.default_url}")
 
-        self.__update_spawn_options(partition_info)
+        if "root_dir" in self.user_options:
+            self.notebook_dir = self.user_options["root_dir"]
```

Comment:
Keeping modification of attributes here, since `options_from_form` is not supposed to have "side effects".